sdgym/synthesizers/base.py

```diff
@@ -59,36 +62,39 @@ class SingleTableBaseline(Baseline, abc.ABC):
     MODALITIES = ('single-table', )
     CONVERT_TO_NUMERIC = False
 
-    def _transform_fit_sample(self, real_data, metadata):
-        ht = rdt.HyperTransformer()
+    def _get_transformed_trained_synthesizer(self, real_data, metadata):
+        self.ht = rdt.HyperTransformer()
         columns_to_transform = list()
         fields_metadata = metadata['fields']
-        id_fields = list()
+        self.id_fields = list()
         for field in fields_metadata:
             if fields_metadata.get(field).get('type') != 'id':
                 columns_to_transform.append(field)
             else:
-                id_fields.append(field)
+                self.id_fields.append(field)
+
+        self.id_field_values = real_data[self.id_fields]
 
-        ht.fit(real_data[columns_to_transform])
-        transformed_data = ht.transform(real_data)
-        synthetic_data = self._fit_sample(transformed_data, metadata)
-        reverse_transformed_synthetic_data = ht.reverse_transform(synthetic_data)
-        reverse_transformed_synthetic_data[id_fields] = real_data[id_fields]
+        self.ht.fit(real_data[columns_to_transform])
+        transformed_data = self.ht.transform(real_data)
+        return self._get_trained_synthesizer(transformed_data, metadata)
+
+    def _get_reverse_transformed_samples(self, data):
+        synthetic_data = self._sample_synthesizer(data)
+        reverse_transformed_synthetic_data = self.ht.reverse_transform(synthetic_data)
+        reverse_transformed_synthetic_data[self.id_fields] = self.id_field_values
         return reverse_transformed_synthetic_data
 
-    def fit_sample(self, real_data, metadata):
-        _fit_sample = self._transform_fit_sample if self.CONVERT_TO_NUMERIC else self._fit_sample
-        if isinstance(real_data, dict):
-            return {
-                table_name: _fit_sample(table, metadata.get_table_meta(table_name))
-                for table_name, table in real_data.items()
-            }
+    def get_trained_synthesizer(self, data, metadata):
+        return self._get_transformed_trained_synthesizer(data, metadata) if (
+            self.CONVERT_TO_NUMERIC) else self._get_trained_synthesizer(data, metadata)
 
-        return _fit_sample(real_data, metadata)
+    def sample_synthesizer(self, synthesizer, n_samples):
```

Comment:
can we add docstrings for these methods? Also, I know it's in the issue, but to me `sample_synthesizer` sounds like you are going to sample a new synthesizer as opposed to sampling from it